tests/store/artifact/test_databricks_artifact_repo.py

```diff
@@ -0,0 +1,137 @@
+                credentials=mock_credentials)
+            write_credentials_mock.return_value = write_credentials_response_proto
+            azure_upload_mock.return_value = None
+            databricks_artifact_repo.log_artifact(test_file.strpath, artifact_path)
+            write_credentials_mock.assert_called_with(MOCK_RUN_ID, expected_location)
+            azure_upload_mock.assert_called_with(mock_credentials, test_file.strpath)
+
+    @pytest.mark.parametrize("artifact_path", [
+        None,
+        'output/',
+        '',
+    ])
+    def test_log_artifacts(self, databricks_artifact_repo, test_dir, artifact_path):
+        with mock.patch(DATABRICKS_ARTIFACT_REPOSITORY + '.log_artifact') as log_artifact_mock:
+            log_artifact_mock.return_value = None
+            databricks_artifact_repo.log_artifacts(test_dir.strpath, artifact_path)
+            artifact_path = artifact_path or ''
+            calls = [mock.call(os.path.join(test_dir.strpath, 'empty-file'), os.path.join(artifact_path, '')),
+                     mock.call(os.path.join(test_dir.strpath, 'test.txt'), os.path.join(artifact_path, '')),
+                     mock.call(os.path.join(test_dir.strpath, 'subdir/test.txt'),
+                               os.path.join(artifact_path, 'subdir'))]
+            log_artifact_mock.assert_has_calls(calls)
+
+    def test_list_artifacts(self, databricks_artifact_repo):
+        with mock.patch(DATABRICKS_ARTIFACT_REPOSITORY + '._call_endpoint') as call_endpoint_mock:
+            list_artifact_response_proto = ListArtifacts.Response(root_uri='', files=LIST_ARTIFACTS_PROTO_RESPONSE)
+            call_endpoint_mock.return_value = list_artifact_response_proto
+            artifacts = databricks_artifact_repo.list_artifacts('test/')
+            assert len(artifacts) == 2
+            assert artifacts[0].path == 'test/a.txt'
+            assert artifacts[0].is_dir is False
+            assert artifacts[0].file_size == 100
+            assert artifacts[1].path == 'test/dir'
+            assert artifacts[1].is_dir is True
+            assert artifacts[1].file_size is 0
+
+            # Calling list_artifacts() on a path that's a file should return an empty list
+            list_artifact_response_proto = ListArtifacts.Response(root_uri='',
+                                                                  files=LIST_ARTIFACTS_SINGLE_FILE_PROTO_RESPONSE)
+            call_endpoint_mock.return_value = list_artifact_response_proto
+            artifacts = databricks_artifact_repo.list_artifacts('a.txt')
+            assert len(artifacts) == 0
+
+    @pytest.mark.parametrize("remote_file_path, local_path", [
+        ('test_file.txt', ''),
+        ('test_file.txt', None),
+        ('output/test_file', None),
+    ])
+    def test_databricks_download_file(self, databricks_artifact_repo, remote_file_path, local_path):
+        with mock.patch(DATABRICKS_ARTIFACT_REPOSITORY + '._get_read_credentials') as read_credentials_mock, \
```

Comment:
Can we verify that the `_get_read_credentials` method is called with the expected run ID? This will indirectly test our run ID parsing.

Reply:
Similarly, it is being checked on line 136.

Reply:
Sorry! Totally missed this. Looks great!